sticky-navbar

Awesomeasma opened the pull request:
The navbar was already modified to stick to the top of the page. I did no further changes.

narayan954 commented:
are you sure ? maybe you didn't get my point, it should be fixed on the top of the screen.
https://user-images.githubusercontent.com/77617189/222951645-a6a7d96b-34fd-45db-a476-584cdf06de21.mp4

narayan954 commented:
this commit was absolutely unnecessary.

Awesomeasma commented:
> are you sure ? maybe you didn't get my point, it should be fixed on the top of the screen. [Watch the video](https://user-images.githubusercontent.com/77617189/222951645-a6a7d96b-34fd-45db-a476-584cdf06de21.mp4)

@narayan954 
 I apologize for that. I have updated the file and pushed updates into the 'sticky-navbar' branch now. Please review it

narayan954 commented on src/index.css:
```diff
@@ -214,34 +216,53 @@
   .scrollTop {
     left: 47.7%;
   }
+  .app__header {
```
I don't think you need to copy paste this property at every query. Find a better solution(remember the DRY or Don't Repeat Yourself principle)

narayan954 reviewed (changes requested):
Hi @Awesomeasma It still doesn't work. Are you sure you fixed it?

Awesomeasma commented:
I am not able to find the reason why it's not working, I have tried to change CSS, but nothing seems to work. Would u mind to tell me where I did wrong.

narayan954 commented:
> I am not able to find the reason why it's not working, I have tried to change CSS, but nothing seems to work. Would u mind to tell me where I did wrong.

position:fixed is the correct configuration, but it'll need some more configuration inside.

narayan954 commented:
And by the way, prior to finalising this pr, when I ask you, please merge all of these commits into a single one, as It's a bad practice to have multiple commits with same name and no valuable changes. And these commits will go into this project history and create confusions. For more info in doing this, you may refer to [this stackoverflow answer](https://stackoverflow.com/questions/5721566/combining-multiple-commits-into-one-prior-to-push)

Awesomeasma commented:
> > I am not able to find the reason why it's not working, I have tried to change CSS, but nothing seems to work. Would u mind to tell me where I did wrong.
> 
> position:fixed is the correct property, but it'll need some more configuration for the elements inside. You may ask me for more hint

Adding padding-top to the body ???

Awesomeasma commented:
> And by the way, prior to finalising this pr, when I ask you, please merge all of these commits into a single one, as It's a bad practice to have multiple commits with same name and no valuable changes. And these commits will go into this project history and create confusions. For more info in doing this, you may refer to [this stackoverflow answer](https://stackoverflow.com/questions/5721566/combining-multiple-commits-into-one-prior-to-push)

Ok sir, I will keep this in mind

Awesomeasma commented:
> > > > I am not able to find the reason why it's not working, I have tried to change CSS, but nothing seems to work. Would u mind to tell me where I did wrong.
> > > 
> > > 
> > > position:fixed is the correct property, but it'll need some more configuration for the elements inside. You may ask me for more hint
> > 
> > 
> > Adding padding-top to the body ???
> 
> yes that works, although I am thinking, whether it should be margin or padding? Both work here, but I'll need to think. Anyways you may do your task with padding.

I would have paid more attention to that, I really took a lot of time to figure that out. Thank u for your patience. 
I have updated the changes in the branch.. please review it.

narayan954 commented on src/index.css:
```diff
@@ -16,19 +16,26 @@
+  
 }
 
 .app__header {
```
give a white background to app header and some margin-right(approx 10px) to logout button

narayan954 commented on src/index.css:
```diff
@@ -16,19 +16,26 @@
   flex-wrap: wrap;
+
+}
+body {
```
header is also a part of body so logically instead of giving padding to body, give app_posts the top padding

Awesomeasma replied:
I am getting a lot of errors, I don't think I have altered 'settings.json' file, but it showed this. I am not sure if I should follow the command shown or not..
![error-vscode](https://user-images.githubusercontent.com/86073340/223298124-6a154205-9f5e-40c8-93db-df307adc372f.png)

Awesomeasma replied:
I think I did it, Please check once..

Awesomeasma replied:
It's showing that there are conflicts in package.json and package-lock.json files, I haven't altered those files at all. I don't understand why it's showing conflicts in those

narayan954 replied:
Okay I'll look into it now

Awesomeasma replied:
Did u resolve it, shall I try again to merge all my 11 commits into one ?

narayan954 replied:
yes you can! :)

narayan954 replied:
Or you know what?,if you're facing problems, you can create a new branch from latest master and put these changes in it and create a pr from that branch only.

Awesomeasma replied:
u mean i should create a new branch to merge my commits into single one ?

narayan954 replied:
> u mean i should create a new branch to merge my commits into single one ?

I mean you should create a new branch from the latest master and fix the sticky navbar issue there. This branch's commit history is quite messed up

narayan954 replied:
> > u mean i should create a new branch to merge my commits into single one ?
> 
> I mean you should create a new branch from the latest master and fix the sticky navbar issue there. This branch's commit history is quite messed up so nuke this branch out of existence

Awesomeasma replied:
Sir, I tried to do what u suggested, but I don't know why I am continously getting conflicts in readme and package.json files, I haven't modified those files at all. Please tell me what to do.

Awesomeasma replied:
Shall I delete all newly-made branches and start fresh ??

narayan954 replied:
> Shall I delete all newly-made branches and start fresh ??

Yes, don't stress too much on this issue.

narayan954 replied:
> Sir, I tried to do what u suggested, but I don't know why I am continously getting conflicts in readme and package.json files, I haven't modified those files at all. Please tell me what to do.

They're arriving because of the merge commits in between (in those merge commits there're different changes and thus they're conflicting in the rebase flow of chronological order) , that's why I am suggesting to remove this branch.

Awesomeasma replied:
I have deleted this branch, and tried to update the changes related to sticky navbar in a new branch, but I see that the css file is already updated with the required changes, there's nothing left to modify.. what do I do now ?

narayan954 replied:
are you sure, css file is already updated? or have you missed something in using the correct branch? Also, if you can run the application, check if it's working fine and you're up to date with the website in production. As a final resort, delete the folder and clone the repository again, then create the branch.

narayan954 reviewed (changes requested):
I have mentioned some minor changes, please do them , then as I said, merge these commits into one and we're successfully done with this issue!
